packages/react-dom-bindings/src/server/ReactDOMServerExternalRuntime.js

```diff
@@ -74,34 +69,49 @@ function installFizzInstrObserver(target /*: Node */) {
 }
 
 function handleNode(node_ /*: Node */) {
-  // $FlowFixMe[incompatible-cast]
-  if (node_.nodeType !== 1 || !(node_ /*: HTMLElement*/).dataset) {
+  if (
+    node_.nodeType !== 1 ||
+    // $FlowFixMe[incompatible-cast]
+    !(node_ /*: HTMLElement*/).dataset
+  ) {
     return;
   }
   // $FlowFixMe[incompatible-cast]
   const node = (node_ /*: HTMLElement*/);
   const dataset = node.dataset;
-  if (dataset['rxi'] != null) {
-    clientRenderBoundary(
-      dataset['bid'],
-      dataset['dgst'],
-      dataset['msg'],
-      dataset['stck'],
-    );
-    node.remove();
-  } else if (dataset['rri'] != null) {
-    // Convert styles here, since its type is Array<Array<string>>
-    completeBoundaryWithStyles(
-      dataset['bid'],
-      dataset['sid'],
-      JSON.parse(dataset['sty']),
-    );
-    node.remove();
-  } else if (dataset['rci'] != null) {
-    completeBoundary(dataset['bid'], dataset['sid']);
-    node.remove();
-  } else if (dataset['rsi'] != null) {
-    completeSegment(dataset['sid'], dataset['pid']);
+
+  if (dataset.hasOwnProperty('ri')) {
     node.remove();
+    switch (dataset.ri) {
```

Comment:
Originally the idea was to encode the instruction in the name of the first argument. That's why it was one argument per instruction since it's fewer bytes to do `data-rxi="arg0"` than `"data-ri="x" data-bid="arg0"`. We should do that instead.

I'm not sure why the instruction attribute wasn't used for arguments previously though.

Reply:
The escaping of the name like `dataset['ri']` is good practice because we should ideally be able to use closure compiler advanced mode and in that case these properties would otherwise be minified. So we've been trying to stick to that as much as possible.

Reply:
sounds good, made them more compact